metaseq/cli/train.py

```diff
@@ -203,10 +203,10 @@ def main(cfg: DictConfig) -> None:
         disable_iterator_cache=True,
     )
 
-    max_epoch = cfg.optimization.max_epoch or math.inf
     train_meter = meters.StopwatchMeter()
     train_meter.start()
-    while epoch_itr.next_epoch_idx <= max_epoch:
+
+    while True:
```

Comment:
Usually it is practice to avoid `while True` anywhere since it relies on other code to stop the loop and it's easy to make mistakes. However, the alternative of splitting logic between loop and validate function is more complex and thus more likely for us to have issues in future.

Also, another reason the while true is not be so bad is because the original could potentially be the same condition when `max-epochs` was not provided / defined.

If not defined 
```
while epoch_itr.next_epoch_idx <= max_epoch
while epoch_itr.next_epoch_idx <= cfg.optimization.max_epoch or math.inf
while epoch_itr.next_epoch_idx <= math.inf
while true
```